src/Approvals/SFA.DAS.Approvals/Application/Cohorts/Queries/GetCohortDetails/GetCohortDetailsQuery.cs

```diff
@@ -21,34 +27,54 @@ public class GetCohortDetailsQueryResult
     {
         public string ProviderName { get; set; }
         public string LegalEntityName { get; set; }
+        public bool HasUnavailableFlexiJobAgencyDeliveryModel { get; set; }
     }
 
     public class GetCohortDetailsQueryHandler : IRequestHandler<GetCohortDetailsQuery, GetCohortDetailsQueryResult>
     {
+        private readonly IDeliveryModelService _deliveryModelService;
         private readonly ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> _apiClient;
         private readonly ServiceParameters _serviceParameters;
+        private readonly IFjaaApiClient<FjaaApiConfiguration> _fjaaClient;
+        private readonly IFjaaService _fjaaService;
 
-        public GetCohortDetailsQueryHandler(ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> apiClient, ServiceParameters serviceParameters)
+        public GetCohortDetailsQueryHandler(IDeliveryModelService deliveryModelService, ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> apiClient, ServiceParameters serviceParameters, IFjaaService fjaaService)
         {
+            _deliveryModelService = deliveryModelService;
             _apiClient = apiClient;
             _serviceParameters = serviceParameters;
+            _fjaaService = fjaaService;
         }
 
         public async Task<GetCohortDetailsQueryResult> Handle(GetCohortDetailsQuery request, CancellationToken cancellationToken)
         {
+            var hasUavailableFjaaDms = false;
+
+            var apiRequest = new GetDraftApprenticeshipsRequest(request.CohortId);
             var cohortRequest = new GetCohortRequest(request.CohortId);
 
-            var cohortResponseTask = await _apiClient.GetWithResponseCode<GetCohortResponse>(cohortRequest);
+            var apiResponseTask = _apiClient.GetWithResponseCode<GetDraftApprenticeshipsResponse>(apiRequest);
+            var cohortResponseTask = _apiClient.GetWithResponseCode<GetCohortResponse>(cohortRequest);
 
+            await Task.WhenAll(apiResponseTask, cohortResponseTask);
 
-            if (cohortResponseTask.StatusCode == HttpStatusCode.NotFound)
+            if (apiResponseTask.Result.StatusCode == HttpStatusCode.NotFound || cohortResponseTask.Result.StatusCode == HttpStatusCode.NotFound)
             {
                 return null;
             }
 
-            cohortResponseTask.EnsureSuccessStatusCode();
+            apiResponseTask.Result.EnsureSuccessStatusCode();
+            cohortResponseTask.Result.EnsureSuccessStatusCode();
+
+            var apprenticeships = apiResponseTask.Result.Body;
+            var cohort = cohortResponseTask.Result.Body;
+
+            var cohortIsOnRegister = await _fjaaService.IsLegalEntityOnFjaaRegister(cohort.AccountLegalEntityId);
 
-            var cohort = cohortResponseTask.Body;
+            if (!cohortIsOnRegister)
+            {
+                hasUavailableFjaaDms = apprenticeships.DraftApprenticeships.Any(a => a.DeliveryModel.Equals(DeliveryModel.FlexiJobAgency));
```

Comment:
This could be changed to:

```HasUnavailableFlexiJobAgencyDeliveryModel = !cohortIsOnRegister && apprenticeships.DraftApprenticeships.Any(a => a.DeliveryModel.Equals(DeliveryModel.FlexiJobAgency))```

to simplify things

Reply:
Neat - refactored